Spec/Auth.swift

```diff
@@ -213,10 +213,11 @@ class Auth : QuickSpec {
                 }
 
                 // RSA4a
-                it("should transition the connection to the FAILED state when the server responds with a token error and there is no way to renew the token") {
+                it("should transition the connection to the disconnected state when the server responds with a token error and there is no way to renew the token") {
```

Comment:
This spec still says that it should transition to Failed state:

> (RSA4a) When a token or tokenDetails is used to instance the library, and no means to renew the token is provided (either an API key, authCallback or authUrl), if the server responds with a token error (401 HTTP status code and an Ably error value 40140 <= code < 40150), then the client library should indicate an error, not retry the request and in the case of the realtime library, transition the connection to the FAILED state

Is the spreadsheet of 1.1 outdated?

Reply:
The spec is correct, and in the RSA4a case you have to transition to `FAILED` because you have no option - there is no means to renew the token, and the token you have doesn't work. The same is true in RTN15h1.

RTN15h2 describes the case where you do have the ability to renew the token, and in this case you go to `DISCONNECTED`. Remember the `DISCONNECTED` state means "I'm currently not connected, but I think I can be connected imminently".

We think this is the definitive text: https://github.com/ably/docs/blob/integration-spec-1-1/content/client-lib-development-guide/features.textile

If the spreadsheet disagrees, then the spreadsheet is wrong.